variables.tf

```diff
@@ -1,87 +1,88 @@
+  description = "The name of the resource group in which to create the resource."
+  nullable    = false
 }
 
-# required AVM interfaces
-# remove only if not supported by the resource
-# tflint-ignore: terraform_unused_declarations
-variable "customer_managed_key" {
-  type = object({
-    key_vault_resource_id = string
-    key_name              = string
-    key_version           = optional(string, null)
-    user_assigned_identity = optional(object({
-      resource_id = string
-    }), null)
-  })
-  default     = null
-  description = <<DESCRIPTION
-A map describing customer-managed keys to associate with the resource. This includes the following properties:
-- `key_vault_resource_id` - The resource ID of the Key Vault where the key is stored.
-- `key_name` - The name of the key.
-- `key_version` - (Optional) The version of the key. If not specified, the latest version is used.
-- `user_assigned_identity` - (Optional) An object representing a user-assigned identity with the following properties:
-  - `resource_id` - The resource ID of the user-assigned identity.
-DESCRIPTION  
+variable "restart_policy" {
+  type        = string
+  description = "The restart policy for the container group."
 }
 
-variable "diagnostic_settings" {
+variable "containers" {
   type = map(object({
-    name                                     = optional(string, null)
-    log_categories                           = optional(set(string), [])
-    log_groups                               = optional(set(string), ["allLogs"])
-    metric_categories                        = optional(set(string), ["AllMetrics"])
-    log_analytics_destination_type           = optional(string, "Dedicated")
-    workspace_resource_id                    = optional(string, null)
-    storage_account_resource_id              = optional(string, null)
-    event_hub_authorization_rule_resource_id = optional(string, null)
-    event_hub_name                           = optional(string, null)
-    marketplace_partner_resource_id          = optional(string, null)
+    image  = string
+    cpu    = number
+    memory = number
+    ports = list(object({
+      port     = number
+      protocol = string
+    }))
+    volumes = map(object({
+      mount_path           = string
+      name                 = string
+      read_only            = optional(bool, false)
+      empty_dir            = optional(bool, false)
+      secret               = optional(map(string), null)
+      storage_account_name = optional(string, null)
+      storage_account_key  = optional(string, null)
+      share_name           = optional(string, null)
+      git_repo = optional(object({
+        url       = optional(string, null)
+        directory = optional(string, null)
+        revision  = optional(string, null)
+      }))
+    }))
+    environment_variables        = optional(map(string), {})
+    secure_environment_variables = optional(map(string), {})
```

Comment:
Because we are including this in here, you would have to mark the whole variable as sensitive. If you can move this out of here into it's own variable you could work around that.